scripts/ci/pr_link_checker.py

```diff
@@ -45,105 +52,123 @@ def get_added_lines_with_links(base_ref: str = "origin/main") -> dict[str, list[
                 env=env,
             )
     except subprocess.CalledProcessError as e:
-        print(f"Error getting git diff: {e}", file=sys.stderr)
+        eprint(f"Error getting git diff: {e}")
         return {}
 
-    lines_by_ext: dict[str, list[str]] = {}
+    lines_by_file: dict[str, list[str]] = {}
     current_file: str | None = None
-    current_ext: str | None = None
 
     for line in result.stdout.split("\n"):
         if line.startswith("+++"):
             # Extract filename from +++ b/path/to/file
             if line.startswith("+++ b/"):
                 current_file = line[6:]  # Remove '+++ b/'
-                current_ext = Path(current_file).suffix.lstrip(".")
-                if current_ext not in lines_by_ext:
-                    lines_by_ext[current_ext] = []
+                if current_file not in lines_by_file:
+                    lines_by_file[current_file] = []
             else:
                 current_file = None
-                current_ext = None
-        elif line.startswith("+") and not line.startswith("+++") and current_ext:
+        elif line.startswith("+") and not line.startswith("+++") and current_file:
             # This is an added line, check if it contains URLs
             line_content = line[1:]  # Remove the '+' prefix
             if (
                 "http://" in line_content
                 or "https://" in line_content
                 or "ftp://" in line_content
                 or "file://" in line_content
+                or re.search(r"\[.+\]\(.+\)", line_content)  # Markdown links [text](url)
```

Comment:
This has high chances to trigger a false positive due to some language construct (e.g. python list of callable: `funcs[i](args)`. How problematic would that be?